### ISSUE TITLE
Pull to refresh

### ISSUE THREAD

**Issue by gudberg** [gorhom/react-native-bottom-sheet#66]
# Feature Request

The ability to pull the bottom sheet down when it is in its default position and trigger reload of Flatlist/Sectionlist. 

## Why it is needed

Very common action to refresh a list by pulling the list down. 

### Code sample
I have tempted something like having three snap points, the lowest for the data fetching.

```
   const handleSheetChange = useCallback(async index => {
      if (index === 0) {
         // shows the ActivityIndicator and sets a state that is used to locking the bottom sheet
         setFetchingScreen(true);
  
       await fetchData()
  
       setFetchingScreen(false);

        // Move the sheet to original place when data has been fetched
         sheetRef.current?.snapTo(1);
      }
   }, []);
```
But it is slow and pretty buggy. Any suggestions of how it would be best to achieve this?

Best regards and thank you for a great library!

**Comment by gudberg** [comment]
I have tempted something like having three snap points, the lowest for the data fetching. 

`
   const handleSheetChange = useCallback(async index => {
      if (index === 0) {
         // shows the ActivityIndicator and sets a state that is used to locking the bottom sheet
         setFetchingScreen(true);
  
       await fetchData()
  
       setFetchingScreen(false);

        // Move the sheet to original place when data has been fetched
         sheetRef.current?.snapTo(1);
      }
   }, []);
`
But it is slow and pretty buggy. Any suggestions of how it would be best to achieve this? 

Best regards and thank you for a great library!

**Comment by gudberg** [comment]
@gorhom is this possibly in some way now with v2?

**Comment by Angelk90** [comment]
@gudberg : Were you able to find a solution?

**Comment by gudberg** [comment]
No, unfortunately not.

**Comment by Angelk90** [comment]
@gudberg : Should I also implement scroll to infinite?

**Comment by gudberg** [comment]
That would be really cool! 
I am using version of reanimated library

**Comment by Angelk90** [comment]
@gudberg : I would like to try to implement it in a project, I would like to integrate the pull to refresh with the scroll to infinite.
Now I open an issues to find out if @gorhom  could help us.

**Comment by gudberg** [comment]
Were you able to get desired result @Angelk90?

**Comment by Angelk90** [comment]
@gudberg:
Try taking a look at the following example: https://snack.expo.io/rEQ_0FpWP

**Comment by gudberg** [comment]
Thank you for this @Angelk90 , 
However,  I am not seeing that this reloads when you pull the sheet down from initial position. Bounce support must be supported I suppose to get that behaviour.

**Comment by Angelk90** [comment]
@gorhom : are there any news?

**Comment by whalemare** [comment]
PTR is really needed feature|

For now, I do PTR for my self in hacky way:

```
 const animatedIndex = useSharedValue(0)

  useAnimatedReaction(
    () => {
      return animatedIndex.value <= -0.05
    },
    (isNeedRefresh) => {
      if (isNeedRefresh) {
        runOnJS(onRefresh)()
      }
    },
    [animatedIndex],
  )

  return (
    <BottomSheet
      animatedIndex={animatedIndex}
      {/**<BottomSheetFlatList ... />*/}
      </BottomSheet>
```

**Comment by gorhom** [comment]
pull to refresh will be available on `v4` #427
